src/index.ts

```diff
@@ -1 +1,535 @@
+
+                    const harOutputPath = temporaryFile({ extension: 'har' });
+
+                    mitmproxyState = {
+                        proc: execa(
+                            'mitmdump',
+                            [
+                                '--mode',
+                                'wireguard',
+                                ...Object.entries(mitmproxyAddons).map(([addonName, addonPath]) => {
+                                    if (!existsSync(addonPath))
+                                        throw new Error(
+                                            `No "${addonName}" addon for mitmproxy found at "${addonPath}".${
+                                                addonName === 'ipcEvents' || addonName === 'harDump'
+                                                    ? ' The mitmproxy capability requires that this addon is present.'
+                                                    : ''
+                                            }`
+                                        );
+                                    return `-s ${addonPath}`;
+                                }),
+                                '--set',
+                                `hardump=${harOutputPath}`,
+                                '--set',
+                                'ipcPipeFd=3',
+                            ],
+                            {
+                                stdio: ['pipe', 'pipe', 'pipe', 'ipc'],
+                            }
+                        ),
+                        harOutputPath,
+                    };
+
+                    await timeout(
+                        Promise.all([
+                            awaitMitmproxyEvent(
+                                mitmproxyState.proc,
+                                (msg) =>
+                                    msg.status === 'proxyChanged' &&
+                                    msg.servers.some((server) => server.type === 'wireguard' && server.is_running)
+                            )
+                                .then((msg) => {
+                                    if (msg.status !== 'proxyChanged') throw new Error('Unreachable.'); // This will never be reached but we use it as a typeguard
+                                    for (const server of msg.servers) {
+                                        if (
+                                            (server.type !== 'wireguard' && !server.wireguard_conf) ||
+                                            mitmproxyState === undefined
+                                        )
+                                            continue;
+                                        mitmproxyState.wireguardConf = server.wireguard_conf;
+                                        return server.wireguard_conf;
+                                    }
+                                    throw new Error('No WireGuard proxy is running.');
```

Comment:
That error sounds like it's the user's fault.

```suggestion
                                    throw new Error('Failed to start mitmproxy: No WireGuard proxy is running.');
```